app/hooks/useEtherBalance.tsx

```diff
@@ -0,0 +1,19 @@
+import useSWR from 'swr/esm/use-swr';
+import { ethIntoCurVal, ZERO_ETH } from '../utils/currency';
+import { useLedgerEthereumWallet } from './useLedgerEthereumWallet';
+
+export default function useEtherBalance() {
+  const ethWallet = useLedgerEthereumWallet();
+  const { data: balance } = useSWR(
+    '/ether/balance',
+    () => ethWallet.getEtherBalance().then(b => ethIntoCurVal(b)),
```

Comment:
We have had a nasty error with this remember? Maybe compose this outside of useSwr :)